File: TenMinsJava/day53.java
// Check if string contains 1 digit, 1 symbol, 1 uppercase letter and 1 lowercase and is more than 10 characters long
package TenMinsJava;

import java.util.Scanner;

public class day53 {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.err.println("Enter a string with 1 digit, 1 symbol, 1 uppercase and 1 lower and more than 10 characters long");
        String s = scan.nextLine();
        char ch[] = s.toCharArray();
        boolean digit = false;
        boolean symbol = false;
        boolean upper = false;
        boolean lower = false;
        boolean len = false;
        for (Character c : ch) {
            if (Character.isDigit(c)) { digit = true; }
            if (!Character.isLetterOrDigit(c)) { symbol = true; }
            if (Character.isUpperCase(c)) { upper = true; }
            if (Character.isLowerCase(c)) { lower = true; }
            if (s.length() > 10) { len = true; }
        }
        if (digit && symbol && upper && lower && len) {
            System.err.println("yeah man");
        }
        else {
            System.err.printf("requirments: \ndigit: %b \nsymbol: %b \nuppercase: %b \nlowercase: %b \ndigit: %b", digit, symbol, upper, lower, len);
        }
    }
}

File: TenMinsPy/day53.py
# Check if string contains 1 digit, 1 symbol, 1 uppercase letter and 1 lowercase and is more than 10 characters long

import string

class day53:
    def __init__(self):
        s = input("Enter a string with 1 digit, 1 symbol, 1 uppercase and 1 lower and more than 10 characters long\n")
        digit = any(ch.isdigit() for ch in s)
        symbol = any(ch in string.punctuation for ch in s)
        upper = any(ch.isupper() for ch in s)
        lower = any(ch.islower() for ch in s)
        leng = len(s) > 10
        if digit and symbol and upper and lower and leng:
            print('yeah man')
        else:
            print(f"requirments: \ndigit: {digit} \nsymbol: {symbol} \nuppercase: {upper} \nlowercase: {lower} \ndigit: {leng}")

day53()